### PR TITLE
chore: make retention period configurable

### PR DESCRIPTION
**Pull request by katyaterletskaya**
### What change does this PR introduce?

Allow the customization of retention periods

### Why was this change needed?

#4570

### CODE REVIEW

**Comment by katyaterletskaya** [comment]
@scopsy @LetItRock @jainpawan21 could you please review? 🙂

File: apps/worker/src/config/env-validator.ts [hunk 1]
```diff
@@ -53,6 +53,15 @@ const validators: { [K in keyof any]: ValidatorSpec<any[K]> } = {
...
   }),
+  NOTIFICATION_RETENTION_DAYS: num({
+    default: 30,
+  }),
+  MESSAGE_GENERIC_RETENTION_DAYS: num({
+    default: 30,
+  }),
+  MESSAGE_IN_APP_RETENTION_DAYS: num({
+    default: 365,
+  }),
```
[the remark this answers is not in the document]
Comment (LetItRock) [reply apps/worker/src/config/env-validator.ts:64]: can we please introduce also the same changes on the `api` side?

**Review by LetItRock — COMMENTED** [review]
sorry for taking this that long... looks good just left you a few small suggestions and we will be ready to go :)

File: apps/worker/src/config/env-validator.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (katyaterletskaya) [reply apps/worker/src/config/env-validator.ts]: @LetItRock done ✅

**Review by LetItRock — APPROVED** [review]
looks good, thank you! 🙌

**Pull request merged**